lib/rendition.js

```diff
@@ -236,6 +220,35 @@ class Rendition {
             console.log("could not determine encoding/charset");
         }
 
+        try {
+            meta[Metadata.REPO_SIZE] = this.size();
+        } catch (err) {
+            console.log("could not determine file size:", err);
+        }
+
+        try {
+            const dimensions = await imageSize(this.path);
+            meta[Metadata.TIFF_IMAGEWIDTH] = dimensions.width;
+            meta[Metadata.TIFF_IMAGEHEIGHT] = dimensions.height;
+        } catch (err) {
+            // The rendition may or may not be an image, so log error for informational purposes
+            console.log(`Error trying to find dimensions, no dimensions found:`, err.message || err);
+        }
+
+        // ensure we don't send null/undefined/non-finite values
+        if(meta[Metadata.TIFF_IMAGEWIDTH] === null 
+            || meta[Metadata.TIFF_IMAGEWIDTH] === undefined
+            || !Number.isFinite(meta[Metadata.TIFF_IMAGEWIDTH])){
+            console.log(`no width found for rendition of type ${meta[Metadata.DC_FORMAT]}`);
+            delete meta[Metadata.TIFF_IMAGEWIDTH];
+        }
+        if(meta[Metadata.TIFF_IMAGEHEIGHT] === null 
```

Comment:
In case width is undefined, null or an Infinite (which is an invalid integer value in our case), delete the property to ensure no null value gets serialized/send.